lib/dependencies/LocalModulesHelpers.js

```diff
@@ -13,11 +14,19 @@ LocalModulesHelpers.addLocalModule = function(state, name) {
 	return m;
 };
 
-LocalModulesHelpers.getLocalModule = function(state, name) {
+LocalModulesHelpers.getLocalModule = function(state, name, namedModule) {
 	if(!state.localModules) return null;
+	if(namedModule && isRelative(name)) {
+		// resolve dependency name relative to the defining named module
+		name = path.join(path.dirname(namedModule), name);
```

Comment:
This doesn't work on windows, because AMD paths use `/` and `path.dirname` uses `\` on windows.